code/core/src/components/components/Button/Button.tsx

```diff
@@ -167,144 +168,159 @@ const StyledButton = styled('button', {
+      }
+      if (size === 'small') {
+        return '0 10px';
+      }
+      if (size === 'medium') {
+        return '0 12px';
+      }
       return 0;
-    }
-    if (padding === 'small' && size === 'small') {
-      return '0 7px';
-    }
-    if (padding === 'small' && size === 'medium') {
-      return '0 9px';
-    }
-    if (size === 'small') {
-      return '0 10px';
-    }
-    if (size === 'medium') {
-      return '0 12px';
-    }
-    return 0;
-  })(),
-  height: size === 'small' ? '28px' : '32px',
-  position: 'relative',
-  textAlign: 'center',
-  textDecoration: 'none',
-  transitionProperty: 'background, box-shadow',
-  transitionDuration: '150ms',
-  transitionTimingFunction: 'ease-out',
-  verticalAlign: 'top',
-  whiteSpace: 'nowrap',
-  userSelect: 'none',
-  opacity: disabled ? 0.5 : 1,
-  margin: 0,
-  fontSize: `${theme.typography.size.s1}px`,
-  fontWeight: theme.typography.weight.bold,
-  lineHeight: '1',
-  background: (() => {
-    if (variant === 'solid') {
-      return theme.base === 'light' ? theme.color.secondary : darken(0.18, theme.color.secondary);
-    }
-
-    if (variant === 'outline') {
-      return theme.button.background;
-    }
-
-    if (variant === 'ghost' && active) {
-      return transparentize(0.93, theme.barSelectedColor);
-    }
-
-    return 'transparent';
-  })(),
-  color: (() => {
-    if (variant === 'solid') {
-      return theme.color.lightest;
-    }
-
-    if (variant === 'outline') {
-      return theme.input.color;
-    }
-
-    if (variant === 'ghost' && active) {
-      return theme.base === 'light' ? darken(0.1, theme.color.secondary) : theme.color.secondary;
-    }
-
-    if (variant === 'ghost') {
-      return theme.textMutedColor;
-    }
-    return theme.input.color;
-  })(),
-  boxShadow: variant === 'outline' ? `${theme.button.border} 0 0 0 1px inset` : 'none',
-  borderRadius: theme.input.borderRadius,
-  // Making sure that the button never shrinks below its minimum size
-  flexShrink: 0,
-
-  '&:hover': {
-    color: variant === 'ghost' ? theme.color.secondary : undefined,
+    })(),
+    height: size === 'small' ? '28px' : '32px',
+    position: 'relative',
+    textAlign: 'center',
+    textDecoration: 'none',
+    transitionProperty: 'background, box-shadow',
+    transitionDuration: '150ms',
+    transitionTimingFunction: 'ease-out',
+    verticalAlign: 'top',
+    whiteSpace: 'nowrap',
+    userSelect: 'none',
+    opacity: disabled && !readOnly ? 0.5 : 1,
+    margin: 0,
+    fontSize: `${theme.typography.size.s1}px`,
+    fontWeight: theme.typography.weight.bold,
+    lineHeight: '1',
     background: (() => {
-      let bgColor = theme.color.secondary;
-
       if (variant === 'solid') {
-        bgColor =
-          theme.base === 'light'
-            ? lighten(0.1, theme.color.secondary)
-            : darken(0.3, theme.color.secondary);
+        return theme.base === 'light' ? theme.color.secondary : darken(0.18, theme.color.secondary);
```

Comment:
Not sure where this branch was started but this looks like potential conflicts with changes made by Michael?

Also the `&& active` just below, I don't think these should still be here?

Does the branch still need rebasing after the a11y merge by any chance?